tests/iio_attr.c

```diff
@@ -166,6 +198,16 @@ static void dump_device_attributes(const struct iio_device *dev,
 		}
 	}
 	if (wbuf) {
+		if (fd) {
+			fprintf(fd, "\t/* Write null terminated string to device attribute: */\n");
```

Comment:
following comment https://github.com/analogdevicesinc/libiio/pull/289/commits/4acfcd1d6408d5ef41d363323cbc1c2e90e83504#r303320445  it looks like some of these blocks could be grouped into a macro/helper function as well;

when diff-ing the code blocks, what differs is quite often some names, and a few variables;
so, whatever the case is, I guess [for a final version] it would be nice to have some sort of

```
static void iio_cgen_dev_attr(fd, const char *op) {
        if (!fd)
               return;
	fprintf(fd, "\t/* %s IIO buffer Attribute, and put result in string */\n"
		    "\tret = iio_device_buffer_attr_%s(dev, \"%s\", buf, sizeof(buf)); \n", op, attr);
	fprintf(fd, "\t/* For other types, use:\n"
		    "\t *   ret = iio_device_buffer_attr_%s_bool(dev, \"%s\", &v_bool);\n", op, attr);
	fprintf(fd, "\t *   ret = iio_device_buffer_attr_%s_double(dev, \"%s\", &v_double);\n", op, attr);
	fprintf(fd, "\t *   ret = iio_device_buffer_attr_%s_longlong(dev, \"%s\", &v_ll);\n", op, attr);
}
```
in this example, `op == "read"` ; but i'm guessing that `op == "write"` would be fairly similar & maybe some parametrization could help

the main argument to maybe doing this [aside from code de-duplication] is reducing copy+paste errors [i've personally done enough of those to hate them enough]

Reply:
helper functions - sure - i wasn't sure what I was going to end up with when I started, and it is all mostly the same.

Will do. Part assumes that the api always is pretty similar (which it is today).  As long as we all like the output (for now).